docs/guidelines/dockerfile-linting.md

```diff
@@ -0,0 +1,79 @@
+# Dockerfile linting
+
+From [hadolint](https://github.com/hadolint/hadolint):
+> A smarter Dockerfile linter that helps you build [best practice](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices) Docker images. The linter parses the Dockerfile into an AST and performs rules on top of the AST. It stands on the shoulders of [ShellCheck](https://github.com/koalaman/shellcheck) to lint the Bash code inside RUN instructions.
+
+## CI/CD
+
+A great way to ensure that your any changes being brought into your repository complies with the best practices is to add the linting on pull requests as a part of the code review
+
+Below is an example workflow that will run linting on our dockerfiles whenever there is a pull request that contains changes to any dockerfiles.
+
+```yml
+name: Running linting on dockerfile
+run-name: ${{ github.actor }} is running linting on dockerfiles
+on:
+  pull_request:
+    branches: [main] # Limit the workflow to only run on pull requests made to the main branch
+    paths:
+      - "**Dockerfile" # Ensure that we trigger on any dockerfile changes
+jobs:
+  DockerfileLinting:
```

Comment:
YAML files tend to use kebab-case in properties names, could `lint-docker:` a good name for this job?

Reply:
Changed it, I've come across many different casings in YAML files, that I admit I was a little blind to what the preferred casing is